python/ray/data/dataset.py

```diff
@@ -2582,7 +2565,7 @@ def make_generator():
                         )
                 elif isinstance(feature_columns, dict):
                     features = {
-                        key: get_df_values(batch[columns])
+                        key: convert_pandas_to_tf_tensor(batch[columns])
```

Comment:
If we are converting everything to tensors and doing the concatenation ourselves, then we don't need to do the `from_generator` down below right?

Instead we can use `tf.data.Dataset.from_tensor_slices` API and we can actually remove the `output_signature` arg from this method, which was one of the user requests.

cc @Yard1 @clarkzinzow

Reply:
We still need to use the `from_generator()` API since we're generating a stream of batches, not slicing a single mono-tensor into batches. To use `from_tensor_slices()`, we'd need to pull the entire `Dataset` into memory before yielding any tensor batches.

Reply:
Got it. But then, do we still require the `output_signature` argument?

Basically, if we are converting to tensorflow tensors ourselves, then I'm wondering if we can remove the `output_signature` arg from the `to_tf` method. Having to specify the `output_signature` was a painpoint that was brought up by a user during dogfooding.

Reply:
Yes that painpoint is understood, however, I don't think that we should remove `output_signature` completely. We should make `output_signature` optional such that it can still be used to cast to different data types, and if not specified, we should infer the types and shapes from the schema and underlying data.

IMO this should happen in a separate PR to keep this diff small and to unblock the examples.